src/coreclr/scripts/genLttngProvider.py

```diff
@@ -88,6 +88,34 @@
         "win:Binary"        :"const BYTE"
         }
 
+monoLttngDataTypeMapping ={
+        #constructed types
+        "win:null"          :" ",
+        "win:Int64"         :"const int64_t",
+        "win:ULong"         :"const uint32_t",
+        "win:count"         :"*",
+        "win:Struct"        :"const uint8_t *",
+        #actual spec
+        "win:GUID"          :"const int32_t",
+        "win:AnsiString"    :"const uint8_t*",
+        "win:UnicodeString" :"const uint8_t*",
+        "win:Double"        :"const double",
+        "win:Int32"         :"const int32_t",
+        "win:Boolean"       :"const bool",
+        "win:UInt64"        :"const uint64_t",
+        "win:UInt32"        :"const uint32_t",
+        "win:UInt16"        :"const uint16_t",
+        "win:UInt8"         :"const uint8_t",
+        "win:Pointer"       :"const size_t",
+        "win:Binary"        :"const BYTE"
+        }
```

Comment:
I haven't tried it yet. I can try building and running the trace and pgo tools in coreCLR with this mapping and see how that works out. We diverged in https://github.com/dotnet/runtime/blob/ac6e77b676068c13878aa1a89bdccfaabde50d35/src/coreclr/scripts/genEventing.py#L44-L106 as well, should we try to unify that into one mapping as well?

Reply:
If everything can use one mapping, that would be great! I'm content with having two different lists if we need it, but reducing duplication is worthy effort.

Reply:
Original kept separate to make sure CoreCLR could continue to use its regular native types, while Mono keep use of C99 types. Some of the CoreCLR types won't work on Mono, so then we would standardize on the Mono types if unified, but since the mapping lives side by side, not sure how much it will benefit to unify them and they probably won't change ones established (only if we add new types into the etw manifest file)

Reply:
I attempted to unify both mappings in genEventing and genLttngProvider, but `ep_char8_t` is causing problems. Perhaps unification can be done in a separate PR